### PR TITLE
Fix javadocs warnings in the sdk-core package

### PR DESCRIPTION
**Pull request by sugmanue**
## Motivation and Context

This change fixes all the current javadoc warnings in the sdk-core package. This was done to avoid introducing new ones in my upcoming changes.

## Modifications

Removed all the current warnings by fixing some incorrect uses of javadocs, such as

* Use of literal < instead of the `&lt;` entity.
* Add imports for unknown classes referenced in `{@link …}`
* Avoid using `@see` when the doc meant to use `{@link …}`
* Avoid links to private fields, instead rephrase the intent

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [x] My change requires a change to the Javadoc documentation
- [x] I have updated the Javadoc documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: core/sdk-core/src/main/java/software/amazon/awssdk/core/traits/Trait.java [hunk 1]
```diff
@@ -16,6 +16,7 @@
 package software.amazon.awssdk.core.traits;
 
 import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.core.SdkField;
```
Comment (L-Applin) [inline core/sdk-core/src/main/java/software/amazon/awssdk/core/traits/Trait.java:19]: Why are we adding that import?

File: core/sdk-core/src/main/java/software/amazon/awssdk/core/SdkNumber.java [hunk 2]
```diff
@@ -27,20 +27,23 @@
...
- * @see #fromBigDecimal(BigDecimal) to create from a BigDecimal.
- * @see #fromBigInteger(BigInteger) to create from a BigInteger.
- * @see #fromDouble(double) to create from a double
- * @see #fromFloat(float) to create from a float.
- * @see #fromLong(long) to create from a long.
- * @see #fromShort(short) to create from a short.
- * @see #fromInteger(int) to create from an integer.
- * @see #fromString(String)  to create from a Stringl,
- * Thus by doing this, this class is able to preserve arbitrary precison of any given number.
+ * <ul>
```
Comment (L-Applin) [inline core/sdk-core/src/main/java/software/amazon/awssdk/core/SdkNumber.java:30]: What was the problem with using `@see`?

Comment (sugmanue) [reply core/sdk-core/src/main/java/software/amazon/awssdk/core/SdkNumber.java:30]: `@see` tags are to generate `See Also` sections generally linking to documents or API outside the current one, not links in general, that's what `{@link …} is designed for usually below the main document, as in 

---
Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.

See Also [Lorem ipsum wikipedia entry](https://en.wikipedia.org/wiki/Lorem_ipsum)

---

Using `@see` here also produces a warning about `warning: Tag {@link}: nested link. If {@link SdkNumber} is expected in a particular number format then its corresponding getter methods can be used.` given that there's leading content below the **See Also** section of the docs.

File: core/sdk-core/src/main/java/software/amazon/awssdk/core/traits/Trait.java (hunk 1 above)
Comment (sugmanue) [reply core/sdk-core/src/main/java/software/amazon/awssdk/core/traits/Trait.java:19]: `SdkField` is being used as a link in the javadocs. Without the import javadoc does not know how to resolve the type and generate the link.

File: core/sdk-core/src/main/java/software/amazon/awssdk/core/SdkNumber.java (hunk 2 above)
Comment (sugmanue) [reply core/sdk-core/src/main/java/software/amazon/awssdk/core/SdkNumber.java:30]: See also the way it is currently rendered [here](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/core/SdkNumber.html).

**Review by L-Applin — APPROVED** [review]

**Pull request merged**